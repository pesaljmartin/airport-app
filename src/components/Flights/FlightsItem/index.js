import React from "react";

import "./index.scss";

const FlightsItem = props => {
  const { departure, origin, destination, detail } = props.flight;
  return (
    <div className="flights-item-wrapper">
      <div>
        <div className="flights-item-wrapper__route">
          <p>Destination: {origin && origin.value}</p>
          <p className="arrow-wrapper">
            <i className="arrow up" />
          </p>
          <p>{destination && destination.value}</p>
        </div>
        <p>Departure: {departure && new Date(departure).toLocaleString()}</p>
      </div>
      <div>
        <p>Flight number: {detail && detail.flight_number}</p>
        <p>
          Boarding time: {detail && new Date(detail.boarding).toLocaleString()}
        </p>
      </div>
      <div>
        <p>Gate: {detail && detail.gate}</p>
        <p>Seat: {detail && detail.seat}</p>
      </div>
    </div>
  );
};

export default FlightsItem;
