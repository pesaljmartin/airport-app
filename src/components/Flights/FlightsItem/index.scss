.flights-item-wrapper {
  padding: 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-radius: 5px;
  display: flex;

  > div:not(:first-child) {
    border-left: 2px solid #000000;
    padding: 0 10px;
  }

  > div:first-child {
    padding-right: 10px;
  }

  > div:not(:last-child) {
    width: 40%;
  }

  &__route {
    display: flex;
    align-items: center;

    .arrow-wrapper {
      padding: 0 8px 0 5px;
    }
  }
}
