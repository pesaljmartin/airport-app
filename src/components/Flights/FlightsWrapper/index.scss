.flights-wrapper {
  max-width: 1024px;
  margin: 0 auto;

  &__sort {
    display: flex;

    &__item {
      display: flex;
      align-items: center;
      padding: 10px;

      &__arrows {
        display: flex;
        flex-direction: column-reverse;
        margin-right: 5px;

        i {
          cursor: pointer;

          &.down {
            transform: rotate(45deg);
          }

          &.up {
            transform: rotate(-135deg);
          }
        }
      }
    }
  }

  p {
    margin: 0 0 5px 0;
  }

  i {
    display: inline-block;
    padding: 3px;
    border: solid #000000;
    border-width: 0 3px 3px 0;
    transform: rotate(-45deg);
  }
}
