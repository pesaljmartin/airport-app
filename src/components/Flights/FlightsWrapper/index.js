import React from "react";

import FlightsItem from "../FlightsItem";
import "./index.scss";

const FlightsWrapper = ({ flights, sortByName, sortByTime }) => {
  return (
    <div className="flights-wrapper">
      <div className="flights-wrapper__sort">
        <div className="flights-wrapper__sort__item">
          <div className="flights-wrapper__sort__item__arrows">
            <p>
              <i className="arrow down" onClick={() => sortByName(false)} />
            </p>
            <p>
              <i className="arrow up" onClick={() => sortByName(true)} />
            </p>
          </div>
          <p>Name</p>
        </div>
        <div className="flights-wrapper__sort__item">
          <div className="flights-wrapper__sort__item__arrows">
            <p>
              <i className="arrow down" onClick={() => sortByTime(false)} />
            </p>
            <p>
              <i className="arrow up" onClick={() => sortByTime(true)} />
            </p>
          </div>
          <p>Time</p>
        </div>
      </div>
      {flights.map(value => {
        return <FlightsItem key={value.segment_id} flight={value} />;
      })}
    </div>
  );
};

export default FlightsWrapper;
