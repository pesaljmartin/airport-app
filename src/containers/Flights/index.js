import React, { useState, useEffect } from "react";

import FlightsWrapper from "../../components/Flights/FlightsWrapper";

const Flights = () => {
  const [flights, setFlights] = useState([]);

  useEffect(() => {
    fetch("./data/index.json")
      .then(response => response.json())
      .then(data => {
        setFlights(data);
      })
      .catch(function(error) {
        console.log(error);
      });
  }, []);

  const sortByName = direction => {
    if (direction) {
      flights.sort((a, b) => {
        return a.origin.value > b.origin.value ? 1 : -1;
      });
    } else {
      flights.sort((a, b) => {
        return a.origin.value > b.origin.value ? -1 : 1;
      });
    }

    setFlights([...flights]);
  };

  const sortByTime = direction => {
    if (direction) {
      flights.sort((a, b) => {
        return a.departure - b.departure;
      });
    } else {
      flights.sort((a, b) => {
        return b.departure - a.departure;
      });
    }

    setFlights([...flights]);
  };

  return <FlightsWrapper {...{ flights, sortByName, sortByTime }} />;
};

export default Flights;
