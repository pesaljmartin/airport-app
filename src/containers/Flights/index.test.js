import React from "react";
import Flights from "./index";
import renderer from "react-test-renderer";

test("Flights comoponent snapshot", () => {
  const component = renderer.create(<Flights />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
