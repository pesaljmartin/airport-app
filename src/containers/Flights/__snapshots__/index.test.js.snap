// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flights comoponent snapshot 1`] = `
<div
  className="flights-wrapper"
>
  <div
    className="flights-wrapper__sort"
  >
    <div
      className="flights-wrapper__sort__item"
    >
      <div
        className="flights-wrapper__sort__item__arrows"
      >
        <p>
          <i
            className="arrow down"
            onClick={[Function]}
          />
        </p>
        <p>
          <i
            className="arrow up"
            onClick={[Function]}
          />
        </p>
      </div>
      <p>
        Name
      </p>
    </div>
    <div
      className="flights-wrapper__sort__item"
    >
      <div
        className="flights-wrapper__sort__item__arrows"
      >
        <p>
          <i
            className="arrow down"
            onClick={[Function]}
          />
        </p>
        <p>
          <i
            className="arrow up"
            onClick={[Function]}
          />
        </p>
      </div>
      <p>
        Time
      </p>
    </div>
  </div>
</div>
`;
